### DISCUSSION TITLE
UI issues v1.6 QT5 (delete rules, dragging the mouse)

### DISCUSSION THREAD

**Discussion by izar3** [evilsocket/opensnitch#1037] [Development]
Hi, This is probably a QT5 issue but heres the issue i'm facing from the last 2 updates of v1.6

When selecting multiple rules (ctrl-a) or dragging the mouse then using the delete key or right-click then delete, it doesn't delete all highlighted items and 99% of the time 1 or 2 remain in the list

I've even had times where i have deleted a rule but its still active in the events tab, like the UI hasn't updated or something

I have tried a fresh install of the OS to see if it fixes the issues but still does it

thanks for you time

on:
KDE 6.1.49-1-MANJARO (64-bit)
Opensnitch v1.6.3-1
python-pyqt5 5.15.9-2

**Comment by gustavo-iniguez-goya** [comment]
Hi @izar3 ,

In my system selecting all visible rules with the mouse deletes the rules correctly. However selecting the rules with ctrl+a deletes only the visible rules but no the rest ones (?), or something weird happens, I need to analyze it.

Anyway, there're behaviour differences among PyQt versions, specially dragging the mouse to select rows.

I'll take a look at it.

**Comment by izar3** [comment]
Any update?

from testing the easiest way to replicate:

1. Go to rules tab on opensnitch UI
2. ctrl+click 2 or more temporary rules
3. press delete key

Some remain, list is broken. Switching to other tab then back to rules fixes the list

**Reply by gustavo-iniguez-goya** [reply]
no yet @izar3 , I haven't had time yet to look into this.

**Reply by gustavo-iniguez-goya** [reply]
ok, so there're three issues here:
 1. dragging the mouse to select rules beyond the visible rows
 2. deleting all the rules selected with ctrl+a
 3. after deleting a rule, the lits is not refreshed
 4. extra: (dragging the mouse to select rows beyond the visible rules, doesn't refresh the list properly, blank rows appear)

I've fixed points 4., 3. (177d67d0a6dc99b58abebb6080589f6e5385d708) and 2. (5fd7da8ef50eeed0746673da36ceddf38d4ee84e) I'll pushed the changes soon.
I haven't managed to fix the point 1 yet, but I'll work on it a little bit more.

(btw, offtopic: izar comes from izarra?)

**Reply by gustavo-iniguez-goya** [reply]
point 4. fixed 174c63c3ffe1cccc83f15d892a0e8c9aec594a9a

The first point requires more work.

**Comment by hashbi** [comment]
Thanks for fixing this. I also was experiencing similar issues

**Comment by izar3** [comment]
This is fixed! <3
